packages/api/cms-api/src/common/guards/external-request-without-header.guard.ts

```diff
@@ -0,0 +1,35 @@
+import { CanActivate, ExecutionContext, Injectable } from "@nestjs/common";
+import { GqlExecutionContext } from "@nestjs/graphql";
+import { Request } from "express";
+
+@Injectable()
+export class ExternalRequestWithoutHeaderGuard implements CanActivate {
+    private headerName: string;
+    private headerValue: string;
+
+    constructor({ headerName, headerValue }: { headerName: string; headerValue: string }) {
+        this.headerName = headerName;
+        this.headerValue = headerValue;
+    }
+
+    private isRequestInternal(request: Request): boolean {
+        // This check is only ok when the application is behind a trusted proxy
+        return !request.header("x-forwarded-for");
```

Comment:
that works in kubernetes where the api pod is behind a haproxy for external requests, and accessed directly for internal requests (BFF), right?

Reply:
Yes exactly. Technically, it is a nginx proxy. But this should work for any trusted proxy.

Reply:
as discussed, enable/disable per boolean in code